### DISCUSSION TITLE
BTest suite & script-coverage

### DISCUSSION THREAD

**Discussion by vpax** [zeek/zeek#3494] [Archive]
Currently when running the main Zeek BTest suite, its configuration includes: `ZEEK_PROFILER_FILE=%(testbase)s/.tmp/script-coverage/XXXXXX`. This generates 100s of MB of data in `.tmp/script-coverage` every run. When doing significant development this quickly turns into GBs.

Chatting with @timwoj we're not sure that this data is actually ever used in practice. In addition, seems for sure it should be an _optional_ testing feature (which could be turned on using a BTest `-a` alternative) rather than the default.

**Comment by timwoj** [comment]
I definitely agree with making this an optional thing instead of a mandatory one.

**Comment by awelzel** [comment]
> Chatting with @timwoj we're not sure that this data is actually ever used in practice. In addition, seems for sure it should be an optional testing feature (which could be turned on using a BTest -a alternative) rather than the default.

It's used in CI:
```
1809 tests successful, 7 skipped
8043/11337 (70.9%) Zeek script statements covered.
...
[#2] tests.m57-long ... ok
all 3 tests successful
3362/9592 (35.1%) Zeek script statements covered.
...
3672/9592 (38.3%) Zeek script statements covered.
```

https://github.com/zeek/zeek/blob/9e9c2e2f32564ce1318beec77ef66cb97abcaa9c/testing/btest/Makefile#L18-L19


No gripe dropping it from the default `[environment]` so it's not impacting local development. Would vote against yet another alternative, however. Instead, setting the environment variable in `ci/test.sh` explicitly should just do? It's also the only place where `make coverage` is invoked.

Think this can go, too (last time we chatted there was no difference anymore): https://github.com/zeek/zeek/blob/9e9c2e2f32564ce1318beec77ef66cb97abcaa9c/ci/test.sh#L77-L84

**Reply by timwoj** [reply]
> It's used in CI

I feel like it's another one of those tools we have available to us but no one actually looks at the data from. These coverage numbers and the C++ coverage numbers are nice to have but not if they're untracked.

> No gripe dropping it from the default [environment] so it's not impacting local development. Would vote against yet another alternative, however. Instead, setting the environment variable in ci/test.sh explicitly should just do? It's also the only place where make coverage is invoked.

I would be ok with this. I think another baseline will just never get run.

**Comment by awelzel** [comment]
Completed with: https://github.com/zeek/zeek/pull/3510
